src/main/java/com/znsio/teswiz/runner/Driver.java

```diff
@@ -603,4 +603,70 @@ public void doubleTap(WebElement element) {
                 .addAction(touch.createPointerUp(PointerInput.MouseButton.LEFT.asArg()));
         appiumDriver.perform(Arrays.asList(clickPosition));
     }
+
+    private Sequence fingerAction(String fingerName, Point locus, int startRadius, int endRadius, double angle, Duration duration) {
+
+        PointerInput finger = new PointerInput(PointerInput.Kind.TOUCH, fingerName);
+        Sequence fingerPath = new Sequence(finger, 0);
+
+        int fingerStartXPoint = (int)Math.floor(locus.x + startRadius * Math.cos(angle)); //converting from polar coordinates to cartesian
+        int fingerStartYPoint = (int)Math.floor(locus.y - startRadius * Math.sin(angle));
+
+        int fingerEndXPoint = (int)Math.floor(locus.x + endRadius * Math.cos(angle));
+        int fingerEndYPoint = (int)Math.floor(locus.y - endRadius * Math.sin(angle));
+
+        fingerPath.addAction(finger.createPointerMove(Duration.ZERO, PointerInput.Origin.viewport(), fingerStartXPoint, fingerStartYPoint))
+                .addAction(finger.createPointerDown(PointerInput.MouseButton.LEFT.asArg()))
+                .addAction(new Pause(finger, Duration.ofMillis(10)))
```

Comment:
why exactly Duration of 10 millis was taken can we make it 1 millis or 1 second?

Reply:
10 ms is small pause added before the finger movement can start, 1 ms will be less and 1 s will be too much, a pause of 10 ms - 100 ms is sufficient for finger movement